src/scripts/genEventing.py

```diff
@@ -620,14 +636,13 @@ def generatePlatformIndependentFiles(sClrEtwAllMan, incDir, etmDummyFile, extern
             providerName = providerNode.getAttribute('name')
             providerSymbol = providerNode.getAttribute('symbol')
 
-            eventpipeProviderCtxName = providerSymbol + "_EVENTPIPE_Context"
-            Clrallevents.write('EXTERN_C __declspec(selectany) EVENTPIPE_TRACE_CONTEXT ' + eventpipeProviderCtxName + ' = { W("' + providerName + '"), 0, false, 0 };\n')
+            if is_windows:
```

Comment:
As a short term measure if it produces the right output I'm fine with it, but these various conditions for if (not) is_windows appear fairly asymmetric with duplicated text which makes me suspicious this python code is factored in a way that might make it more error prone to work with. It could be I'm just missing the pattern after my eyes glaze over in 6 hours of code review though. My suggestion is if you agree the factoring is weird, file an issue for us to revisit this. If you don't think its weird we can sit later this week when I my mind is fresh and we can walk through it.

Reply:
It is weird but I wanted to get the filtering work in before the snap happens.... I filed https://github.com/dotnet/coreclr/issues/25388 to track that.